src/index.js

```diff
@@ -20,15 +20,78 @@ function createElement(tagName, props, ...children) {
   return element;
 }
 
-function render() {
+const numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 0];
+const operators = ['+', '-', '*', '/', '='];
+
+function render({
+  leftNumber = 0,
+  operator = '+',
+  rightNumber = 0,
+  nextOperator = '+',
+}) {
+  function calculate({
+    calculatedLeftNumber = 0,
+    calculatedOperator = '+',
+    calculatedRightNumber = 0,
+    nextCalculatedOperator = '+',
+  }) {
+    const operateCase = {
+      '+': calculatedLeftNumber + calculatedRightNumber,
+      '-': calculatedLeftNumber - calculatedRightNumber,
+      '*': calculatedLeftNumber * calculatedRightNumber,
+      '/': calculatedLeftNumber / calculatedRightNumber,
+      '=': calculatedRightNumber,
+    };
+    return {
+      leftNumber: operateCase[calculatedOperator],
+      operator: nextCalculatedOperator,
+      rightNumber: 0,
+      nextOperator: '+',
+    };
+  }
+
+  function handleNumberClick(e) {
+    render({
+      leftNumber,
+      operator,
+      rightNumber: rightNumber * 10 + Number(e.target.value),
+      nextOperator,
+    });
+  }
+
+  function handleOperatorClick(e) {
+    const result = calculate({
+      calculatedLeftNumber: leftNumber,
+      calculatedOperator: operator,
+      calculatedRightNumber: rightNumber || leftNumber,
+      nextCalculatedOperator: e.target.value,
+    });
+    render(result);
+  }
+
   const element = (
     <div>
       <p>간단 계산기</p>
+      <p>{rightNumber || leftNumber}</p>
+      <p>
+        {numbers.map((number) => (
+          <button type="button" value={number} onClick={handleNumberClick}>
+            {number}
+          </button>
+        ))}
+      </p>
+      <p>
+        {operators.map((oper) => (
+          <button type="button" value={oper} onClick={handleOperatorClick}>
+            {oper}
+          </button>
```

Comment:
```suggestion
        {operators.map((operator) => (
          <button type="button" value={operator} onClick={handleClickOperator}>
            {operator}
          </button>
```

`oper`와 같이 줄여서 사용하는건 좋지 않습니다. 물론 충분히 문맥상 유추 가능하지만, 본인만이 생각하는 의미를 동료들에게 이해되지 않을 수 있습니다. 최대한 명확하게 코드를 작성해주세요 :) 

그리고 핸들러 함수의 네이밍도 `handleClick~~`과 같이 작성하는게 좋을 것 같습니다.
- https://developer.mozilla.org/en-US/docs/Web/Guide/Events/Event_handlers
- https://reactjs.org/docs/handling-events.html
- https://blog.sonim1.com/220

Reply:
넵 반영했습니다!!